app/models/manageiq/providers/base_manager/metrics_collector_worker.rb

```diff
@@ -1,10 +1,16 @@
 class ManageIQ::Providers::BaseManager::MetricsCollectorWorker < MiqQueueWorkerBase
+  include MiqWorker::ReplicaPerWorker
+
   require_nested :Runner
 
   include PerEmsTypeWorkerMixin
 
   self.required_roles = ["ems_metrics_collector"]
 
+  def self.supports_container?
+    true
+  end
```

Comment:
Yeah, we could probably move this into the `MiqWorker` modules in some way rather than having it implemented explicitly in each worker.

Alternatively I could just flip the default implementation in `miq_worker.rb` and override where we still don't want to start a worker as a container which might provide more useful information at  this point. What do you think?

Reply:
I think either is fine, and it is probably simpler to only have it in one spot.   Though, if you do invert it in `miq_worker.rb`, I think maybe a disclaimer comment above the method might be worth while, just so those passing through know it is disabled on certain workers.  With the other approach, since you would be adding it to things like `MiqWorker::ReplicaPerWorker` and including it in the worker class, I think that would be more easily assumed.